src/sentry/tasks/integrations/github/pr_comment.py

```diff
@@ -318,3 +325,43 @@ def github_comment_reactions():
             continue
 
         metrics.incr("github_pr_comment.comment_reactions.success")
+
+
+# TODO(adas): Change the client typing to allow for multiple SCM Integrations
+def safe_for_comment(
+    gh_client: GitHubAppsClient, repository: Repository, pull_request: PullRequest
+) -> bool:
+    try:
+        pullrequest_resp = gh_client.get_pullrequest(
+            repo=repository.name, pull_number=pull_request.key
+        )
+    except ApiError as e:
+        if e.json and RATE_LIMITED_MESSAGE in e.json.get("message", ""):
+            metrics.incr(
+                OPEN_PR_METRIC_BASE.format(key="api_error"),
+                tags={"type": "gh_rate_limited", "code": e.code},
+            )
+        elif e.code == 404:
+            metrics.incr(
+                OPEN_PR_METRIC_BASE.format(key="api_error"),
+                tags={"type": "missing_gh_pull_request", "code": e.code},
+            )
+        else:
+            metrics.incr(
+                OPEN_PR_METRIC_BASE.format(key="api_error"),
+                tags={"type": "unknown_api_error", "code": e.code},
+            )
+        raise e
```

Comment:
we're not currently raising an `ApiError` for rate-limited or issue locked errors for merged PR comments. do we want to do the same thing here?

Reply:
Hmm getting an ApiError for being rate limited i think is a good idea. We want to know if we're calling github too much since that affects other features. For other cases, logging the exception and returning False makes sense.

Reply:
oh i see. since this will be run quite often it's probably more indicative of the actual limit if we raise `ApiError` here?